data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParsingExceptions.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import java.io.PrintStream;
+import java.util.List;
+
+/**
+ * @since 1.3
+ *
+ * Exception thrown by {@link ParseTreeParser} if ANTLR parse emits error events.
+ */
+public class ParsingExceptions extends Exception {
+    private final List<Throwable> causes;
+
+    public ParsingExceptions(final String message, final List<Throwable> causes) {
+        super(message);
+        this.causes = causes;
+    }
+
+    @Override
+    public void printStackTrace(final PrintStream s) {
```

Comment:
Why do we need to override this method?

Reply:
The default implementation is only designed to recursively print exceptions that have a single cause. `ParseExceptions` may have multiple causes that should be printed.

Reply:
But AFAIK this print is just a side effect that will not be invoked when we log the error, e.g. LOG.error(e). If that is the case, it will not be useful in production.

Reply:
Thanks @chenqi0805, after some testing you're correct. Overriding printStackTrace did not change the message printed. I've refactored the custom exception to work as expected and added some unit tests to verify.